packages/x-data-grid/package.json

```diff
@@ -52,7 +52,7 @@
     "@mui/x-internals": "workspace:*",
     "clsx": "^2.1.1",
     "prop-types": "^15.8.1",
-    "reselect": "^4.1.8"
+    "reselect": "^5.1.0"
```

Comment:
> Switches `createSelector` to use a new `weakMapMemoize` method as the default memoizer

@romgrk Do you know if switching to [`weakMapMemoize`](https://reselect.js.org/api/weakMapMemoize) impacts the data grid's performance?

Reply:
Thought your commit reverted to v4 behavior, I now see that it doesn't for `createSelector`. I have opened this PR to restore it: https://github.com/mui/mui-x/pull/14410

Reply:
I even forgot I pushed any commits to this PR 😅